Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts

```diff
@@ -489,7 +489,7 @@ export class ResourceGroup {
         return new Promise<void>((resolve, reject) => {
             console.log(tl.loc("StartingValidation"));
             deployment.properties["mode"] = "Incremental";
-            armClient.deployments.validate(this.taskParameters.resourceGroupName, this.createDeploymentName(), deployment, (error, result, request, response) => {
+            armClient.deployments.validate(this.taskParameters.resourceGroupName, this.taskParameters.deploymentName || this.createDeploymentName(), deployment, (error, result, request, response) => {
```

Comment:
the 'or' logic can be moved to when `this.deploymentMode` is assigned.

Reply:
@bishal-pdMSFT do you mean treat the deployment name as a property (as mode) or use the createDeploymentName() method if null? 

eg. 
_this.taskParameters.deploymentName = this.taskParameters.deploymentName || this.createDeploymentName();_

Reply:
I meant the latter i.e. 
_this.taskParameters.deploymentName = this.taskParameters.deploymentName || this.createDeploymentName();_